toolchain/check/check_unit.cpp

```diff
@@ -394,6 +396,18 @@ auto CheckUnit::ProcessNodeIds() -> bool {
   return true;
 }
 
+auto CheckUnit::CheckRequiredDeclarations() -> void {
+  for (const auto& function : context_.functions().array_ref()) {
+    if (!function.first_owning_decl_id.has_value() &&
+        function.extern_library_id == this->context_.sem_ir().library_id()) {
```

Comment:
Can you add a cross-package test, such as:

```
// --- cross_package_extern.carbon

package OtherPackage library "[[@TEST_NAME]]";

extern library "cross_package" fn Extern();

// --- cross_package.carbon

library "[[@TEST_NAME]]";

// This call causes the function to be imported.
fn F() { OtherPackage.Extern(); }
```

Note here that the library name matches but the package name does not.

I suspect this is going to warn. Maybe this should be a TODO that the code needs to verify the function comes from the same package? (seems fine to split out PRs)

Reply:
Added test with a TODO, it's currently incorrectly emitting diagnostics. Can I retrieve the package name for the function declaration in CheckRequiredDeclarations?